cli/azd/internal/repository/app_init.go

```diff
@@ -137,11 +137,23 @@ func (i *Initializer) InitFromApp(
 		if prj.Language == appdetect.Java {
 			var hasKafkaDep bool
 			var hasSpringCloudAzureDep bool
-			for _, dep := range prj.AzureDeps {
-				if eventHubs, ok := dep.(appdetect.AzureDepEventHubs); ok && eventHubs.UseKafka {
+			for depIndex := range prj.AzureDeps {
+				dep := &prj.AzureDeps[depIndex]
+				if eventHubs, ok := (*dep).(appdetect.AzureDepEventHubs); ok && eventHubs.UseKafka {
 					hasKafkaDep = true
+					springBootVersion := eventHubs.SpringBootVersion
+
+					if springBootVersion == appdetect.UnknownSpringBootVersion {
+						var err error
+						springBootVersion, err = promptSpringBootVersion(i.console, ctx)
+						if err != nil {
+							return err
+						}
+						eventHubs.SpringBootVersion = springBootVersion
+						prj.AzureDeps[depIndex] = eventHubs
```

Comment:
I don't think we can just use `(*dep).(appdetect.AzureDepEventHubs)` to update directly, because it's actually a copy of the element, not the actual element.